File: displayframes.rb
require 'rubygems'
require 'bundler/setup'

require 'fileutils'
require 'taglib'

# now that files have been converted to mp3's, update the ID3 tags
puts 'Detecting MP3 files...'
Dir.glob('*.mp3') do |file|
  puts '*' * 20
  puts "File: #{File.basename(file)}"

    TagLib::MPEG::File.open(file) do |mp3_file|
      tag = mp3_file.id3v2_tag
      tag.frame_list.each do |f|
        puts f.frame_id
        puts f.description if f.respond_to?(:description)
        puts f.to_string
        if f.frame_id == 'APIC'
          puts f.description
          puts f.mime_type
          puts f.text_encoding
          puts f.type
        end
      end
      puts
    end
end

File: displayframes.go
package main

import (
	"fmt"
	"log"
	"path/filepath"

	"github.com/bogem/id3v2"
)

func main() {
	// addPicture()

	fmt.Println("Detecting MP3 files...")
	files, _ := filepath.Glob("*.mp3")
	for i := 0; i < len(files); i++ {
		fmt.Println("********************")
		fmt.Println("File: " + files[i] + ":")

		// Open file and find tag in it
		tag, err := id3v2.Open(files[i])
		if err != nil {
			log.Fatal("error opening file", err)
		}

		defer tag.Close()

		frameList := tag.AllFrames()
		for key, value := range frameList {
			fmt.Println(key)
			fmt.Println(value)
		}
	}
}
